client/src/components/utils/Post/index.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Icon } from 'antd';
+import { Link, BrowserRouter as Router } from 'react-router-dom';
+import './style.css'
+
+export default function Post({ onClick, postTitle, id, postType }) {
+  return (
+    <Router>
+      <Link to={`/post/${id}`}>
+        <div className='post'>
+          <p className='post--title'>
+            {postTitle.length > 30 ? postTitle.substring(0, 31) : postTitle}
+          </p>
+          <div>
+            <Icon
+              id={id}
+              onClick={() => onClick(id)}
```

Comment:
@IsraaSulaiman You can remove this inline arrow function by changing this component into a class component, which wouldn't be too much effort hopefully.
Then you'd have:
```
export default class Post extends React.component {
  onClick = () => {
    this.props.onClick( this.props.id );
  }

  render() {
    ...
    <Icon onClick={this.onClick}
    ...
  }
}
```

Removing the inline arrow functions inside functional components or render functions is a good idea because they are a little inefficient - these inline arrow functions are recreated every time the component changes and is re-rendered, so they use more memory than they need to because there are unnecessary copies of them.